### PR TITLE
parse/format ints and floats

### PR DESCRIPTION
**Pull request by TurkeyMan**
Need these functions to manipulate RGB format strings...

### CODE REVIEW

File: src/wg/util/format.d [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * Format an integer as a string.
+ */
+char[] formatInt(I)(I i, ref char[21] output, uint minSize = 0) @safe pure nothrow @nogc
+{
+    assert(minSize <= 20, "Too many digits!");
+
+    uint numDigits = 0;
+    if (i == 0)
+    {
```
Comment (drug007) [inline src/wg/util/format.d]: I'd add `if(minSize == 0) minSize = 1;` here (or minSize set to 1 by default) else if `i` equals to zero output will be zero length.
```
unittest
{
        char[21] output;
        assert(formatInt(0, output) == "0");
}
```

Comment (TurkeyMan) [reply src/wg/util/format.d]: Nice catch!

File: src/wg/util/traits.d [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module wg.util.traits;
+
+// traits
+enum bool isFloatingPoint(T) = __traits(isFloating, T);
+
+enum bool isSigned(T) = __traits(isArithmetic, T) && !__traits(isUnsigned, T);
+
+enum bool isClassOrStructPtr(T) = is(T == class) || (is(T == U*, U) && is(U == struct));
```
Comment (dayllenger) [inline src/wg/util/traits.d]: `interface`?

Comment (TurkeyMan) [reply src/wg/util/traits.d]: I guess that should be fine...

**Comment by TurkeyMan** [comment]
Thanks!

**Comment by drug007** [comment]
Thank you for your job!

**Pull request merged**